### ISSUE TITLE
Crashes when launching the app for the first time from 6.7.0

### ISSUE THREAD

**Issue by NUmeroAndDev** [AppsFlyerSDK/appsflyer-android-sdk#37]
- Crashes when launching the app for the first time from 6.7.0.
  - No crash when downgrading to 6.6.1.
- Happens when `minifyEnabled` is true.
  - Proguard sets https://dev.appsflyer.com/hc/docs/install-android-sdk#proguard-rules.  

#### Libraries
`com.appsflyer:af-android-sdk:6.7.0@aar`

#### Proguard

```
-keep class com.appsflyer.** { *; }
-keep public class com.android.installreferrer.** { *; }
```

### Crashlog

```
FATAL EXCEPTION: pool-32-thread-1
java.lang.NoSuchMethodError: No direct method <init>(Ljava/util/Collection;)V in class Lcom/appsflyer/internal/k; or its super classes (declaration of 'com.appsflyer.internal.k' appears in /data/app/~~0xfNjkn093xYJ3R3UPCWCw==/-_BB99_BhRJTQSn_hmeenNw==/base.apk)
    at com.appsflyer.internal.t.AFInAppEventType(:288)
    at com.appsflyer.internal.t.<init>(:152)
    at com.appsflyer.internal.t.AFInAppEventType(:293)
    at com.appsflyer.internal.t.<init>(:152)
    at com.appsflyer.internal.o$a.AFInAppEventType(:51)
    at com.appsflyer.internal.o.AFInAppEventParameterName(:37)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.appsflyer.internal.bb.valueOf(:1)
    at com.appsflyer.internal.bx.values(:11)
    at com.appsflyer.internal.bw.AFInAppEventType(:6)
    at com.appsflyer.internal.bp$4.run(:23)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
    at java.lang.Thread.run(Thread.java:1012)
```

**Comment by ge-af** [comment]
Hi @NUmeroAndDev.

We think this may be a problem with R8 Optimizer that may be obfuscating our SDK classes even though you are using -keep ProGuard rules.
Due to the complexity of this issue,

We would kindly ask you to submit a support ticket to support@appsflyer.com in order to get optimal support.
When submitting the ticket, please specify your AppsFlyer sign-up (account) email, app ID, reproduction steps, code snippets, logs, the links you used, and any additional relevant information.

**Comment by leearmstrong** [comment]
We are also seeing this with 6.7.0 and can confirm the Proguard rules are in place.

**Comment by jdwarraich** [comment]
Im Facing same issue with 6.7.0

**Comment by MrChips-debug** [comment]
@leearmstrong I'm receiving same issue with 6.7.0 when minify enabled, while 6.6.1 works fine for me

**Comment by leearmstrong** [comment]
Ok. We've downgraded too and I have emailed support. They aren't being too fast though!

**Comment by MrChips-debug** [comment]
@leearmstrong Thanks for your response :)

**Comment by ge-af** [comment]
Hi all,

We are sorry if there has been any delay in our messaging, we are expecting high volume in Support these days.
Please file support ticket and we will be more than happy to provide the best solution based on the issue you've encountered.
It may be that some of the issues mentioned here are not the same and we will investigate each case to its full.

The above issue seems to be with Google's R8 Optimizer and the Mobile R&D team are on working on it with top priority. I'll update once a solution will be available but I can't guarantee any ETA at the moment.

**Comment by ge-af** [comment]
R&D has found the culprit for this issue and seems to resolve it.
This is going to be fixed in the next official release of the Android SDK (v6.8.0), it should be released in the next couple of weeks.

**Comment by ge-af** [comment]
This was fixed in latest Android SDK 6.8.0 version.
Marking as closed.
